test/nebulex_redis_adapter/stats_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule NebulexRedisAdapter.StatsTest do
+  use ExUnit.Case, async: true
+
+  defmodule Cache do
+    use Nebulex.Cache,
+      otp_app: :nebulex,
+      adapter: NebulexRedisAdapter
+  end
+
+  setup do
+    {:ok, pid} = Cache.start_link(stats: true)
+
+    on_exit(fn ->
+      :ok = Process.sleep(100)
+      if Process.alive?(pid), do: Cache.stop(pid)
+    end)
+
+    {:ok, cache: Cache, name: Cache}
+  end
+
+  describe "c:NebulexRedisAdapter.stats/1" do
+    test "returns valid %Stats{}" do
+      refute Cache.get("stats")
```

Comment:
I see the problem, the thing is at this moment we don't know if Redis has some entries already, so let's do something:

```elixir
size = Cache.size()

assert Cache.put("stats", "stats") == :ok
assert Cache.get("stats") == "stats"
assert Cache.put("stats", "stats") == :ok
assert Cache.take("stats") == "stats"
refute Cache.get("stats")
assert Cache.put_new("stats", "stats")
assert Cache.replace("stats", "stats stats")
assert Cache.flush() == size + 1

assert stats = Cache.stats()
assert stats.measurements.evictions == size + 2
assert stats.measurements.hits == 2
assert stats.measurements.misses == 2
assert stats.measurements.writes == 3
assert stats.measurements.updates == 1
```

Reply:
Is there a failure in CI? Tests all pass locally, but that's with a clean start of redis.

Reply:
Actually, we should use `Cache.count_all/0`, since `Cache.stats/0` will be deprecated. And no need to specify a database.

Reply:
Already using `Cache.count_all/0` and `Cache.delete_all/0` 😉